opencog/atomese/atomese_utils/constants.h

```diff
@@ -14,6 +14,7 @@ const static Handle compressed_value_key = createNode(NODE, "*-AS-MOSES:SchemaCo
 
 const static ValuePtr true_value(createLink(TRUE_LINK));
 const static ValuePtr false_value(createLink(FALSE_LINK));
+extern combo::type_node t_output_type;
```

Comment:
Why do you need to have `t_output_type` as global variable?

Reply:
The `t_output_type`  is a required field in `REGRESSION`, `REGRESSION` is used by all `table_problem` types. Hence, its not convenient to declare `t_output_type` as private for each struct. most `table_problem` types have no use of `t_output_type` , that's why a default value is given  to them. We can declare `t_output_type` with a default value in `REGRESSION` and that can solve this but I haven't tried doing that with `#define` statements . Maybe that can solve it